models/user.go

```diff
@@ -174,6 +174,18 @@ func CreateUser(email, username, password string) error {
 	return nil
 }
 
+// UserFactory will create and return an instance of a user
+func UserFactory(email, username, password string) (User, error) {
+	u := User{}
+	err := createUser(email, username, password)
+	if err != nil {
+		return u, err
+	}
+	err = u.FindByEmail(email)
+
+	return u, err
+}
+
```

Comment:
This code probably looks hella familiar, except now when you call this method you get back both the error and the newly created user, like you were asking about.